### PR TITLE
Add string comparison in reflect.DeepEqual

### LINKED ISSUE #69686: "Observed a panic" error logs in kubelet every 5s
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:
After a freshly installed cluster, the following error appears every 5s in the `kubelet` logs:
```
Oct 11 17:47:03 serverlab61 kubelet[19325]: E1011 17:47:03.196677   19325 runtime.go:66] Observed a panic: reflect.unexportedTypePanic{(*reflect.rtype)(0x36e9300), (*reflect.rtype)(0x33d41e0), (*reflect.rtype)(0x38043e0), (*reflect.rtype)(0x3499fc0), (*reflect.rtype)(0x3800ee0), (*reflect.rtype)(0x37a0fe0), (*reflect.rtype)(0x3047580)} (an unexported field was encountered, nested like this: *checkpoint.remoteConfigMap -> checkpoint.remoteConfigMap -> *v1.NodeConfigSource -> v1.NodeConfigSource -> *v1.ConfigMapNodeConfigSource -> v1.ConfigMapNodeConfigSource -> string)
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:72
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
Oct 11 17:47:03 serverlab61 kubelet[19325]: /usr/local/go/src/runtime/asm_amd64.s:573
Oct 11 17:47:03 serverlab61 kubelet[19325]: /usr/local/go/src/runtime/panic.go:502
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:96
Oct 11 17:47:03 serverlab61 kubelet[19325]: /usr/local/go/src/runtime/asm_amd64.s:573
Oct 11 17:47:03 serverlab61 kubelet[19325]: /usr/local/go/src/runtime/panic.go:502
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:96
Oct 11 17:47:03 serverlab61 kubelet[19325]: /usr/local/go/src/runtime/asm_amd64.s:573
Oct 11 17:47:03 serverlab61 kubelet[19325]: /usr/local/go/src/runtime/panic.go:502
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:96
Oct 11 17:47:03 serverlab61 kubelet[19325]: /usr/local/go/src/runtime/asm_amd64.s:573
Oct 11 17:47:03 serverlab61 kubelet[19325]: /usr/local/go/src/runtime/panic.go:502
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:96
Oct 11 17:47:03 serverlab61 kubelet[19325]: /usr/local/go/src/runtime/asm_amd64.s:573
Oct 11 17:47:03 serverlab61 kubelet[19325]: /usr/local/go/src/runtime/panic.go:502
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:96
Oct 11 17:47:03 serverlab61 kubelet[19325]: /usr/local/go/src/runtime/asm_amd64.s:573
Oct 11 17:47:03 serverlab61 kubelet[19325]: /usr/local/go/src/runtime/panic.go:502
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:96
Oct 11 17:47:03 serverlab61 kubelet[19325]: /usr/local/go/src/runtime/asm_amd64.s:573
Oct 11 17:47:03 serverlab61 kubelet[19325]: /usr/local/go/src/runtime/panic.go:502
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:96
Oct 11 17:47:03 serverlab61 kubelet[19325]: /usr/local/go/src/runtime/asm_amd64.s:573
Oct 11 17:47:03 serverlab61 kubelet[19325]: /usr/local/go/src/runtime/panic.go:502
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:219
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:186
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:183
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:186
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:183
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:186
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:183
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go:243
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubeletconfig/controller.go:312
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubeletconfig/controller.go:278
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubeletconfig/controller.go:141
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
Oct 11 17:47:03 serverlab61 kubelet[19325]: /workspace/anago-v1.11.3-beta.0.71+a4529464e4629c/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:79
Oct 11 17:47:03 serverlab61 kubelet[19325]: /usr/local/go/src/runtime/asm_amd64.s:2361
```

**What you expected to happen**:
No error logs as it is a freshly installed cluster

**How to reproduce it (as minimally and precisely as possible)**:
New fresh cluster with only `kube-system` namespace

**Anything else we need to know?**:
```
# kubectl get nodes -owide
NAME          STATUS   ROLES    AGE   VERSION   INTERNAL-IP    EXTERNAL-IP   OS-IMAGE                  KERNEL-VERSION                  CONTAINER-RUNTIME
serverlab61   Ready    master   1h    v1.11.3   10.138.132.7   <none>        Oracle Linux Server 7.5   4.1.12-112.16.4.el7uek.x86_64   docker://18.6.1
serverlab62   Ready    master   1h    v1.11.3   10.138.132.8   <none>        Oracle Linux Server 7.5   4.1.12-112.16.4.el7uek.x86_64   docker://18.6.1
serverlab63   Ready    master   1h    v1.11.3   10.138.132.9   <none>        Oracle Linux Server 7.5   4.1.12-112.16.4.el7uek.x86_64   docker://18.6.1
```

**Environment**:
- Kubernetes version (use `kubectl version`):
```
# kubectl version
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.1", GitCommit:"4ed3216f3ec431b140b1d899130a69fc671678f4", GitTreeState:"clean", BuildDate:"2018-10-05T16:46:06Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.2", GitCommit:"bb9ffb1654d4a729bb4cec18ff088eacc153c239", GitTreeState:"clean", BuildDate:"2018-08-07T23:08:19Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration: On-premises (HP G7)
```
# uname -a
Linux serverlab61 4.1.12-112.16.4.el7uek.x86_64 #2 SMP Mon Mar 12 23:57:12 PDT 2018 x86_64 x86_64 x86_64 GNU/Linux
```
- OS (e.g. from /etc/os-release):
```
# cat /etc/os-release
NAME="Oracle Linux Server"
VERSION="7.5"
ID="ol"
VERSION_ID="7.5"
PRETTY_NAME="Oracle Linux Server 7.5"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:oracle:linux:7:5:server"
HOME_URL="https://linux.oracle.com/"
BUG_REPORT_URL="https://bugzilla.oracle.com/"

ORACLE_BUGZILLA_PRODUCT="Oracle Linux 7"
ORACLE_BUGZILLA_PRODUCT_VERSION=7.5
ORACLE_SUPPORT_PRODUCT="Oracle Linux"
ORACLE_SUPPORT_PRODUCT_VERSION=7.5
```
- Kernel (e.g. `uname -a`):
```
# uname -a
Linux serverlab61 4.1.12-112.16.4.el7uek.x86_64 #2 SMP Mon Mar 12 23:57:12 PDT 2018 x86_64 x86_64 x86_64 GNU/Linux
```
- Install tools: `kubeadm`

### PR DESCRIPTION
**Pull request by Chenditang**
**What this PR does / why we need it**:
Add string comparison in reflect.DeepEqual

**Which issue(s) this PR fixes** *(optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged)*:
Fixes #69686

**Special notes for your reviewer**:

**Release note**:
```
NONE
```

### CODE REVIEW

**Comment by idealhack** [comment]
cc @kubernetes/sig-api-machinery-pr-reviews

File: staging/src/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go [hunk 1]
```diff
@@ -174,6 +174,8 @@ func (e Equalities) deepValueEqual(v1, v2 reflect.Value, visited map[visit]bool,
 			}
 		}
 		return true
+	case reflect.String:
```
Comment (liggitt) [inline staging/src/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go]: A test case demonstrating what this is fixing would be good. We deep compare lots of things with strings already, it's unclear why this case is needed (and if it is needed, why cases for all primitive types are not also required)

**Pull request closed without merging**